javascript/ui.js

```diff
@@ -427,6 +427,47 @@ async function browseModels(){
     }
 }
 
+async function searchModel({model_type, searchValue}) {
+    const response = await fetch(`/sd_extra_networks/update_page?model_type=${model_type}&page=1&search_value=${searchValue}&page_size=10&need_refresh=false`, {
+        method: "GET", cache: "no-cache"});
+    return await response.json()
+}
+
+async function getModelFromUrl() {
+
+    // get model form url
+    const urlParam = new URLSearchParams(location.search);
+    const checkpointModelValueFromUrl = urlParam.get('checkpoint').toLowerCase();
+
+    document.cookie = `selected_checkpoint_model=${checkpointModelValueFromUrl}`;
+    const keyMapModelType = {
+        "checkpoint": "checkpoints",
+        "lora": "lora",
+        "ti": "textual_inversion",
+        "hn": "hypernetworks"
+    }
+
+    for (const [key, value] of urlParam.entries()) {
+        if (keyMapModelType[key]) {
+            const { model_list, allow_negative_prompt } = await searchModel({ model_type: keyMapModelType[key], searchValue: value.toLowerCase() })
+            if (model_list.length === 0) {
+                notifier.alert(`${keyMapModelType[key]} ${value} not found`, {
+                    labels: {
+                        alert: 'Model not Found'
+                    }
+                })
+            } else {
+                // checkpoint dont need to replace text
+                if (key !== 'checkpoint') {
+                    cardClicked('txt2img', eval(model_list[0].prompt), allow_negative_prompt);
+                }
+            }
+        }
+    }
```

Comment:
You probably want to use `notifier.asyncBloack(Promise.all([searchModel(), ...]))` to wait for all model search to return to allow user to start generating images.